File: Card Gaem/Card Game By James/js/Hand.js
// File: js/Hand.js
class Hand extends Deck {
    constructor() {
        super();  // Calls the constructor of the Deck class
        this.cards = [];  // Player's hand is initially empty
    }

    addCards(cards) {
        this.cards = this.cards.concat(cards);
    }

    get handString() {
        return this.cards.map(card => card.cardString).join(', ');
    }
}

File: Java game (first js to j conv.)/Hand.java
import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

class Hand extends Deck {
    private List<Card> cards;

    public Hand() {
        super();  // Calls the constructor of the Deck class
        this.cards = new ArrayList<>();  // Player's hand is initially empty
    }

    public void addCards(List<Card> cards) {
        this.cards.addAll(cards);
    }

    public String getHandString() {
        return this.cards.stream()
                         .map(card -> card.getCardString())
                         .collect(Collectors.joining(", "));
    }
}

